tests/unit/Service/SttServiceTest.php

```diff
@@ -33,19 +59,97 @@ public function testDummy() {
+		// case '(not set)', folder 'Application::REC_FOLDER' does not exist
+		$this->doCase('(not set)', Application::REC_FOLDER, false);
+
+		// case '(not set)', folder 'Application::REC_FOLDER' does exist
+		$this->doCase('(not set)', Application::REC_FOLDER . ' 1', true);
+
+		// case 'Application::REC_FOLDER' folder 'Application::REC_FOLDER' does exist
+		$this->doCase(Application::REC_FOLDER, Application::REC_FOLDER, true);
+
+		// null userId
+		$this->expectException(\InvalidArgumentException::class);
+		$this->assertEquals('ok', $this->service->transcribeFile('', true, null));
+	}
+
+	private function doCase(
+		string $configValue,
+		string $finalFolderName,
+		bool $folderExists,
+		string $existingFolderName = Application::REC_FOLDER,
+	) {
+		$this->setUp();
+
+		$userFolder = $this->createMock(Folder::class);
+		$recFolder = $this->createMock(Folder::class);
+		$audioFile = $this->createMock(File::class);
+
+		$this->rootFolder->method('getUserFolder')->willReturn($userFolder);
+
+		$userFolder->method('nodeExists')->willReturnCallback(function (
+			string $name,
+		) use ($existingFolderName, $folderExists) {
+			return $name === $existingFolderName && $folderExists;
 		});
+		$userFolder->method('get')->with($finalFolderName)->willReturn($recFolder);
+		$userFolder->method('newFolder')->willReturn($recFolder);
 
-		$userId = 'dummy';
-		$audioContent = 'badumtss';
+		$recFolder->method('getName')->willReturn($finalFolderName);
+		$recFolder->method('newFile')->willReturnCallback(function (
+			string $filename,
+			mixed $data,
+		) use (&$audioFile) {
+			$audioFile->method('getName')->willReturn($filename);
+			$audioFile->method('fopen')->willReturn($data);
+			return $audioFile;
+		});
+
+		$this->config
+			->method('getAppValue')
+			->with(Application::APP_ID, 'stt_folder', '(not set)')
+			->willReturn($configValue)
+		;
+		$this->config
+			->method('setAppValue')
+			->willReturnCallback(function (string $appId, string $key, string $value) use (&$finalFolderName) {
+				if ($appId !== Application::APP_ID || $key !== 'stt_folder') {
+					throw new \InvalidArgumentException('Invalid arguments to setAppValue');
+				}
+				if ($value !== $finalFolderName) {
+					throw new \InvalidArgumentException('Invalid value to setAppValue');
+				}
+			})
+		;
+
+		$testFileContents = file_get_contents(static::TEST_FILE);
+		$this->manager->method('transcribeFile')->with($audioFile)->willReturn($testFileContents);
 
-		$receivedContent = $this->service->transcribeAudio($audioContent, $userId);
+		$this->assertEquals('ok', $this->service->transcribeAudio(static::TEST_FILE, true, 'dummy'));
 
-		$this->assertEquals($audioContent, $receivedContent);
+		$this->assertEquals($testFileContents, $this->service->transcribeAudio(static::TEST_FILE, false, 'dummy'));
+		$audioFileHandle = $audioFile->fopen('rb');
+		$this->assertEquals(fread($audioFileHandle, filesize(static::TEST_FILE)), $testFileContents);
```

Comment:
Is this line necessary? It seems you're just testing whether your mock works here, no?